.github/workflows/release-python.yml

```diff
@@ -9,101 +9,92 @@ concurrency:
+
     steps:
-      - uses: actions/checkout@v2
-      - name: Setup Python
-        uses: actions/setup-python@v2
+      - name: Checkout pymongoarrow
+        uses: actions/checkout@v2
         with:
-          python-version: ${{ matrix.python-version }}
-          cache: 'pip'
-          cache-dependency-path: '**/setup.cfg'
+          fetch-depth: 0
+
+      - uses: actions/setup-python@v2
+
+      - name: Install cibuildwheel
+        run: python -m pip install "cibuildwheel>=2.4,<3"
+
       - name: Build wheels
         working-directory: ./bindings/python
         shell: bash
-        run: |
-          ./release.sh
+        env:
+          CIBW_BUILD: ${{ matrix.python }}-${{ matrix.buildplat[1] }}
+        run: python -m cibuildwheel --output-dir wheelhouse
+
       - uses: actions/upload-artifact@v2
         with:
-          name: pymongoarrow-${{ matrix.python-version }}-${{ matrix.os }}-wheel
-          path: ./bindings/python/dist/*.whl
+          name: ${{ matrix.python }}-${{ startsWith(matrix.buildplat[1], 'macosx') && 'macosx' || matrix.buildplat[1] }}
+          path: ./bindings/python/wheelhouse/*.whl
           if-no-files-found: error
-  test-non-linux-wheels:
-    needs: build-non-linux-wheels
-    runs-on: ${{ matrix.os }}
-    strategy:
-      matrix:
-        os: [macos-10.15, windows-latest]
-        python-version: [3.7, 3.8, 3.9, "3.10"]
-    name: Test CPython ${{ matrix.python-version }}-${{ matrix.os }}
+
+  make_sdist:
+    name: Make SDist
+    runs-on: ubuntu-latest
     steps:
-      - name: Setup Python
-        uses: actions/setup-python@v2
+      - uses: actions/checkout@v2
         with:
-          python-version: ${{ matrix.python-version }}
-      - name: Download a previously created wheel
-        uses: actions/download-artifact@v2
+          fetch-depth: 0
+
+      - uses: actions/setup-python@v2
         with:
-          name: pymongoarrow-${{ matrix.python-version }}-${{ matrix.os }}-wheel
-      - name: Test wheel
-        shell: bash
+          # Build sdist on lowest supported Python
+          python-version: '3.7'
+
+      - name: Build SDist
+        working-directory: ./bindings/python
         run: |
-          python -m pip install -U pip
-          python -m pip install *.whl
-          python -c "from pymongoarrow.lib import process_bson_stream"
-  # Linux
-  build-manylinux-wheels:
-    runs-on: ubuntu-latest
-    strategy:
-      matrix:
-        # Cannot have '/' in artifact names, so hardcode 'quay.io/pypa/'
-        container: ['manylinux2010_x86_64', 'manylinux2014_x86_64']
-        python-version: ['cp37-cp37m', 'cp38-cp38',
-                         'cp39-cp39', 'cp310-cp310']
-    name: Build CPython ${{ matrix.python-version }}-${{ matrix.container }}
-    steps:
-      - uses: actions/checkout@v2
-      - name: Build wheels
+          set -ex
+          python -m pip install -U pip build
+          export LIBBSON_INSTALL_DIR="$(pwd)/libbson"
+          ./build-libbson.sh
```

Comment:
Can you remind me why it's needed? I just want to make sure we're not including ".so" files and such in the sdist build.

Reply:
I verified there or no binary files in our published sdist.  The build fails because we eagerly add all of the flags.  We could avoid this by using an `Extension` class, which would only get used on install and wheel build.

Reply:
Thanks, the current approach SGTM no need to change it.